meson.build

```diff
@@ -0,0 +1,39 @@
+# SPDX-License-Identifier: Apache-2.0
+
+project('rbmc_data_sync', 'cpp', 'c',
+  default_options: [
+    'buildtype=debugoptimized',
+    'warning_level=3',
+    'werror=true',
+    'cpp_std=c++23',
+    ],
+  meson_version: '>=1.3.0',
+  version: '1.0.0',
+  )
+
+data_sync_config_dir = get_option('datadir') + '/phosphor-data-sync/config/data_sync_list/'
+
+
+foreach json_file_name : get_option('data_sync_list')
+
+    #check whether the json file given in the list exist and if so, install the same
+    json_file = files('config/data_sync_list/' + json_file_name+'.json')
+
+    install_data(
+        json_file,
+        install_dir : data_sync_config_dir
+    )
+
+endforeach
+
+#auto generate a config file with required build time configurations
+conf_data = configuration_data()
+
+conf_data.set_quoted('DATA_SYNC_CONFIG_DIR', get_option('prefix')+'/'+ data_sync_config_dir)
```

Comment:
Please use `description` argument to add some details which will added as comment in the generated header file.

Reply:
Done